### PR TITLE
add 'k8s.io/code-generator' dependency file

### PR DESCRIPTION
**Pull request by dylandee**
just like example in offical repo

https://github.com/kubernetes/sample-controller/blob/master/hack/tools.go

add this file to make sure 'k8s.io/code-generator' would not be wiped out by tpying 'go mod tidy'

### CODE REVIEW

**Comment by FillZpp** [comment]
@dylandee You can put this file in `hack/` and remove the printf line in `scripts/generate_client.sh`

**Comment by dylandee** [comment]
> @dylandee You can put this file in `hack/` and remove the printf line in `scripts/generate_client.sh`

Ok, done

File: scripts/generate_client.sh [hunk 1]
```diff
@@ -1,6 +1,5 @@
 #!/usr/bin/env bash
 
-printf 'package hack\nimport "k8s.io/code-generator"\n' > ./hack/hold.go
 go mod vendor
 retVal=$?
 rm -f ./hack/hold.go
```
Comment (FillZpp) [inline scripts/generate_client.sh:5]: Also remove this line please...

Comment (dylandee) [reply scripts/generate_client.sh]: ok, removed

**Review by FillZpp — APPROVED** [review]
LGTM

**Pull request merged**